routes/api.js

```diff
@@ -1,10 +1,74 @@
 const express = require('express');
+const jwt    = require('jsonwebtoken');
+const config = require('../config/settings');
 const router = express.Router();
+const models = require('../database/models');
+
+const localUser = models.user_account;
+const progress = models.progress;
+
+router.use(function(req, res, next) {
+
+    // check header or url parameters or post parameters for token
+    let token = req.body.token || req.query.token || req.headers['x-access-token'];
+
+    // decode token
+    if (token) {
+        // Removing quotes in the token
+        token = token.substring(1, token.length-1);
+
+        // verifies secret and checks exp
+        jwt.verify(token, config.secretKey, function(err, decoded) {
+            if(err) {
+                return res.json({ success: false, message: 'Failed to authenticate token.' });
+            } else {
+                // if everything is good, save to request for use in other routes
+                req.decoded = decoded;
+                next();
+            }
+        });
+
+    } else {
+        // if there is no token
+        // return an error
+        return res.status(403).send({
+            success: false,
+            message: 'No token provided.'
+        });
+    }
+});
 
 /* GET api listing. */
 router.get('/getUserInfo', (req, res) => {
     res.status(200).json({user: req.user});
 });
 
+router.get('/getProgressStatus', (req, res) => {
+    localUser.find({where: {
+        email: req.user.email
+    }}, {raw: true})
+        .then(data => {
+            if(!data) {
+                return res.status(200).json({info:'This account does not exist'});
+            }
+            progress.find({where: {
+                user_id: data.id
+            }}, {raw: true})
+                .then(progressData => {
+                    if(!progressData) {
+                        return res.status(200).json({info:'No data found'});
+                    }
+                    const response = {stage: progressData.stage, activity: progressData.activity};
+                    return res.status(200).json(response);
+                })
+                .catch(function(err) {
+                    return res.status(500).json({error:'Something went wrong'});
```

Comment:
'Something went wrong' isn't very informative for debugging purposes, especially since this phrase appears in multiple places - do you know which one it's referring to & resolve it accordingly?